codegen/apt/src/main/java/io/helidon/codegen/apt/AptTypeFactory.java

```diff
@@ -161,6 +161,10 @@ public static Optional<TypeName> createTypeName(TypeElement element, TypeMirror
      * @return the associated type name instance
      */
     public static Optional<TypeName> createTypeName(Element type) {
+        if (type == null) {
```

Comment:
This actually violates Helidon design (we only support non-null parameters - always). 
We can merge it, I will create a follow up to fix this.

Reply:
Shall I add a comment here that this was added only to get ECJ running?

Reply:
I have added a comment as to why this check was added.